src/test/java/teammates/test/cases/ui/browsertests/StudentCourseJoinConfirmationPageUiTest.java

```diff
@@ -233,4 +233,14 @@ private LoginPage createCorrectLoginPageType(String pageSource) {
             throw new RuntimeException(e);
         }
     }
+
+    // continuously ask BackDoor to get the key until a legit key is returned
+    private String getKeyFromBackDoor(String courseId, String studentEmail) {
+        String key = "[BACKDOOR_STATUS_FAILURE]";
+        while (key.startsWith("[BACKDOOR_STATUS_FAILURE]")) {
+            key = BackDoor.getKeyForStudent(courseId, studentEmail);
+        }
+        return key;
+    }
```

Comment:
`Retry forever` is a bit risky. Limit to a fixed number of tries. Also, use a delay between tries?